src/entity/Bullet.java

```diff
@@ -1,74 +1,69 @@
 package entity;
 
 import java.awt.Color;
-
 import engine.DrawManager.SpriteType;
 
 /**
  * Implements a bullet that moves vertically up or down.
  * 
- * @author <a href="mailto:RobertoIA1987@gmail.com">Roberto Izquierdo Amo</a>
- * 
+ * @author ...
  */
 public class Bullet extends Entity {
 
-	/**
-	 * Speed of the bullet, positive or negative depending on direction -
-	 * positive is down.
-	 */
-	private int speed;
+    /**
+     * Speed of the bullet, positive or negative depending on direction - positive is down.
+     */
+    private double speed;
 
-	/**
-	 * Constructor, establishes the bullet's properties.
-	 * 
-	 * @param positionX
-	 *            Initial position of the bullet in the X axis.
-	 * @param positionY
-	 *            Initial position of the bullet in the Y axis.
-	 * @param speed
-	 *            Speed of the bullet, positive or negative depending on
-	 *            direction - positive is down.
-	 */
-	public Bullet(final int positionX, final int positionY, final int speed) {
-		super(positionX, positionY, 3 * 2, 5 * 2, Color.WHITE);
-
-		this.speed = speed;
+    private static final int FPS = 60;
+    /**
+     * Constructor, establishes the bullet's properties.
+     * 
+     * @param positionX Initial position of the bullet in the X axis.
+     * @param positionY Initial position of the bullet in the Y axis.
+     * @param isEnemy   If true, the bullet belongs to the enemy; otherwise, it belongs to the player.
+     */
+    public Bullet(final int positionX, final int positionY, final boolean isEnemy) {
```

Comment:
The usages of the constructor should be updated to reflect this change. However, I would recommend using the old constructor so that various bullet speeds can be used by the player and enemies.  The speed can be used to determine whether it is an enemy or not.

Reply:
Also, bullets are sometimes reused via the [BulletPool](https://github.com/brewcoua/Invaders-SDP/blob/feat/bullet/src/entity/BulletPool.java). You may have to change the color to a protected type so that the bullet colors are properly changed when they are being recycled.